src/auth/auth.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Body, Controller, Post, Req, UseGuards } from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { CreateUserDto } from 'src/user/dto';
+import { LoginUserDto } from 'src/user/dto/loginUserDto';
+import { Public } from './decorator';
+import { Role } from 'src/roles/role.enum';
+import { RootAdmin } from './guard';
+
+@Controller('auth')
+export class AuthController {
+  constructor(private authService: AuthService) {}
+
+  @Public()
+  @Post('register')
+  async register(
+    @Body() createUserDto: CreateUserDto,
+    @Req() request: Request,
+  ) {
+    try {
+      const url = request.url;
+
+      return await this.authService.register(
+        createUserDto,
+        url.includes('admin') ? [Role.ADMIN] : [Role.USER],
```

Comment:
You don't really need this check here, because when you make a request to register/admin,another protected handler will be called and on a public route there should be no way to register an admin.